File: Easy/Max Sum Subarray of size K/max-sum-subarray-of-size-k.cpp
//{ Driver Code Starts
#include<bits/stdc++.h> 
using namespace std; 

// } Driver Code Ends
class Solution{   
public:
    long maximumSumSubarray(int K, vector<int> &arr , int N){
        // code here 
        long sum = 0;
        long maxSum = 0;
        
        for(int i=0; i<K; i++)
            sum += arr[i];
            
        int i = K;
        maxSum = sum;
        while(i < N) {
            sum = sum - arr[i-K] + arr[i];
            maxSum = max(maxSum, sum);
            i++;
        }
        
        return maxSum;
    }
};

//{ Driver Code Starts.
int main() 
{ 
    int t;
    cin>>t;
    while(t--)
    {
        int N,K;
        cin >> N >> K;;
        vector<int>Arr;
        for(int i=0;i<N;++i){
            int x;
            cin>>x;
            Arr.push_back(x);
        }
        Solution ob;
        cout << ob.maximumSumSubarray(K,Arr,N) << endl;
    }
    return 0; 
} 
// } Driver Code Ends

File: Difficulty: Easy/Max Sum Subarray of size K/max-sum-subarray-of-size-k.java
class Solution {
    public int maxSubarraySum(int[] arr, int k) {
        int n = arr.length;
        int maxSum = 0, windowSum = 0;
        for (int i = 0; i < k; i++) {
            windowSum += arr[i];
        }
        maxSum = windowSum;
        for (int i = k; i < n; i++) {
            windowSum += arr[i] - arr[i - k];
            if (windowSum > maxSum) maxSum = windowSum;
        }
        return maxSum;
    }
}